packages/core/src/consumed-thing.ts

```diff
@@ -659,7 +683,7 @@ export default class ConsumedThing extends TD.Thing implements IConsumedThing {
 
         const content = await client.invokeResource(form, input);
         // infer media type from form if not in response metadata
-        if (!content.type) content.type = form.contentType;
+        if (!content.type) content.type = form.contentType ?? "application/json";
```

Comment:
```suggestion
        if (content.type == null) {
            content.type = form.contentType ?? "application/json";
        }
```

Reply:
I am not sure if the suggested changes is identical. JS `null` and `undefined` lead to the same boolean while the suggestion is purely about `null`.

Reply:
I also only realized it due to stricter null checks in the linter for node-code, where this check is the only comparison which allows for using `==` instead of `===` :)

Reply:
(Overall, the handling of nullish values is definitely not the most intuitive language feature of JS...)

Reply:
Generally, I avoid using `==` just because it confuses people. I'm wondering why should we use your proposal?

Reply:
Oh, sorry, I think I might have internalized [this eslint rule](https://github.com/typescript-eslint/typescript-eslint/blob/v4.33.0/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md) (which is currently enforced in node-coap) a bit too much. I thought it was a requirement of a stricter tsconfig, but that is not actually the case.

The rationale for using `foo == null` (which is only supposed to be used for this particular null check, otherwise using `==` is discouraged for good reasons) is that it potentially contributes to more type safety, as it does not become true for the empty string, empty arrays and objects, and 0 (in contrast to `!foo`). However, it does potentially require more accompanying type checks to not let these value falls through (which is actually a potential source of new bugs, as I had to experience).

The change was just a suggestion, though, so we can simply mark the thread as resolved :) However, for the future it might be worth considering to have even stricter linting rules to enforce additional type safety.